src/arch/cc_light/cc_light_resource_manager.h

```diff
@@ -161,9 +185,20 @@ class qwg_resource_t : public resource_t
         return true;
     }
 
-    void reserve(size_t op_start_cycle, ql::gate * ins, std::string & operation_name,
-        std::string & operation_type, std::string & instruction_type, size_t operation_duration)
+    void reserve(size_t op_start_cycle, ql::gate * ins, const ql::quantum_platform & platform)
     {
+        std::string operation_name(ins->name);
+        std::string operation_type("cc_light_type");
+        size_t      operation_duration = std::ceil( static_cast<float>(ins->duration) / platform.cycle_time);
+        JSON_ASSERT(platform.instruction_settings, ins->name, ins->name);
+        if ( !platform.instruction_settings[ins->name]["cc_light_instr"].is_null() )
+        {
+            operation_name = platform.instruction_settings[ins->name]["cc_light_instr"];
```

Comment:
Consider creating functions get_operation_name/get_operation_type because there are many repetitions of identical code